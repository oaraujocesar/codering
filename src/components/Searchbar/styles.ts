import styled from 'styled-components'

export const Wrapper = styled.div`
  position: relative;
  width: 100%;
  max-width: 763px;
`

export const Input = styled.input<{ autocomplete: string }>`
  width: 100%;
  outline: none;
  padding: 9px 28px;
  border-radius: 20px;
  border: 1px solid transparent;
  color: ${(props) => props.theme.textColorGrey};
  background: ${(props) => props.theme.shadeOne};
  border: 2px solid transparent;

  &:focus {
    border: 2px solid ${(props) => props.theme.primaryColor};
  }
`

export const Button = styled.button`
  top: 0;
  right: 0;
  border: none;
  height: 100%;
  outline: none;
  display: flex;
  cursor: pointer;
  padding: 4px 20px;
  position: absolute;
  align-items: center;
  border-radius: 0 20px 20px 0;
  background: ${(props) => props.theme.primaryColor};

  &:hover {
    filter: brightness(0.9);
  }
`
