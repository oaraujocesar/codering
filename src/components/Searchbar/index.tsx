import { ChangeEvent } from 'react'

import { useTheme } from 'styled-components'

import SearchIcon from '../Icons/Search'

import * as S from './styles'

interface ISearch {
  placeholder: string
  value?: string
  change?: (text: string) => void
}

const Searchbar = ({ placeholder, value, change }: ISearch) => {
  const theme = useTheme()

  return (
    <S.Wrapper>
      <S.Input
        type="text"
        name="searchbar"
        id="searchbar"
        value={value}
        autocomplete="off"
        onChange={({ currentTarget }: ChangeEvent<HTMLInputElement>) =>
          change && change(currentTarget.value)
        }
        placeholder={placeholder}
      />
      <S.Button role="search">
        <SearchIcon fill={theme?.backgroundColor} />
      </S.Button>
    </S.Wrapper>
  )
}

export default Searchbar
