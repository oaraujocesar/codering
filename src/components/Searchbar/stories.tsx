import { Story, Meta } from '@storybook/react/types-6-0'

import { theme } from '../../styles/theme'
import { ThemeProvider } from 'styled-components'

import Searchbar from '.'

export default {
  title: 'Searchbar',
  component: Searchbar,
  argTypes: {
    currentTheme: { control: { type: 'radio', options: ['light', 'dark'] } }
  }
} as Meta

export const Basic: Story = (args) => {
  const currentTheme = args.currentTheme === 'dark' ? theme.dark : theme.light

  return (
    <ThemeProvider theme={currentTheme}>
      <Searchbar
        placeholder="Digite a tecnologia ou palavras-chave..."
        {...args}
      />
    </ThemeProvider>
  )
}
