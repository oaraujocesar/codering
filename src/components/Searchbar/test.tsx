import { render, screen } from '@testing-library/react'

import Searchbar from '.'

describe('<Searchbar />', () => {
  it('Should render the searchbar', () => {
    const { container } = render(
      <Searchbar placeholder="Digite a tecnologia ou palavras-chave..." />
    )

    expect(
      screen.getByPlaceholderText(/Digite a tecnologia ou palavras-chave.../i)
    ).toBeInTheDocument()

    expect(screen.getByRole('search')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
