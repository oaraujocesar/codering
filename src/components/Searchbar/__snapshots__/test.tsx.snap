// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searchbar /> Should render the searchbar 1`] = `
.c0 {
  position: relative;
  width: 100%;
  max-width: 763px;
}

.c1 {
  width: 100%;
  outline: none;
  padding: 9px 28px;
  border-radius: 20px;
  border: 1px solid transparent;
  border: 2px solid transparent;
}

.c1:focus {
  border: 2px solid;
}

.c2 {
  top: 0;
  right: 0;
  border: none;
  height: 100%;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 4px 20px;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0 20px 20px 0;
}

.c2:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

<div
  class="c0"
>
  <input
    class="c1"
    id="searchbar"
    name="searchbar"
    placeholder="Digite a tecnologia ou palavras-chave..."
    type="text"
    value=""
  />
  <button
    class="c2"
    role="search"
  >
    <svg
      class=""
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.26421 26.9973L12.9602 21.3333C14.8255 22.6259 17.1068 23.1739 19.3557 22.8695C21.6045 22.5651 23.6581 21.4303 25.1126 19.6883C26.5671 17.9463 27.3172 15.7232 27.2155 13.4562C27.1137 11.1891 26.1675 9.0421 24.5628 7.43743C22.9581 5.83275 20.8111 4.88649 18.5441 4.78475C16.277 4.683 14.0539 5.43313 12.3119 6.88761C10.5699 8.34209 9.43516 10.3957 9.13074 12.6445C8.82632 14.8934 9.37429 17.1747 10.6669 19.04L4.97088 24.736C4.67125 25.036 4.50296 25.4427 4.50296 25.8667C4.50296 26.2907 4.67125 26.6973 4.97088 26.9973C5.11921 27.1519 5.29724 27.2748 5.49429 27.3588C5.69134 27.4428 5.90334 27.4862 6.11755 27.4862C6.33176 27.4862 6.54375 27.4428 6.7408 27.3588C6.93785 27.2748 7.11589 27.1519 7.26421 26.9973ZM18.1335 6.93333C19.5048 6.93333 20.8453 7.33997 21.9855 8.10181C23.1257 8.86366 24.0143 9.94649 24.5391 11.2134C25.0639 12.4803 25.2012 13.8744 24.9337 15.2193C24.6661 16.5642 24.0058 17.7996 23.0362 18.7693C22.0665 19.7389 20.8311 20.3993 19.4862 20.6668C18.1412 20.9343 16.7472 20.797 15.4803 20.2722C14.2134 19.7475 13.1305 18.8588 12.3687 17.7186C11.6068 16.5784 11.2002 15.238 11.2002 13.8667C11.203 12.0287 11.9344 10.2668 13.2341 8.96718C14.5337 7.66753 16.2956 6.93615 18.1335 6.93333Z"
        fill=""
      />
    </svg>
  </button>
</div>
`;
